### PR TITLE
Perseus34x/fix refactor alu

### PR DESCRIPTION
**Pull request by perseus34x**
## Motivation

Refactor core modules and fix RotateChip. The implementation now conforms to MachineAir, and the missing rotate op wiring is added. 

## Solution

- A missing inclusion for the rotate operation was added (ff3edfbaa).
- New tests were added, including more edge cases for the rotate functionality (74cfad5dc) and a new test for the LtChip (9380de1aa).

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rwasm/machine/src/alu/rotate/mod.rs [hunk 1]
```diff
@@ -149,9 +149,9 @@ where
         let local = main.row_slice(0);
         let local: &RotateCols<AB::Var> = (*local).borrow();
 
-        // ---------------------------------------------------------------------
+        // ---------------------------------------------------------------------\
```
Comment (djadjka) [inline crates/rwasm/machine/src/alu/rotate/mod.rs]: do you need this '\\' ?

Comment (perseus34x) [reply crates/rwasm/machine/src/alu/rotate/mod.rs]: no
